printserver/print_systems/cups.py

```diff
@@ -155,13 +149,23 @@ def get_printers(self, printer_selector: PrinterSelector) -> list[PrinterDetails
                     pass
                 else:
                     continue
-            if not printer["printer-make-and-model"].lower().startswith(
-                printer_selector.model_prefix.lower()
-            ) or not printer["printer-info"].lower().startswith(
-                printer_selector.name_prefix.lower()
+
+            if (
+                printer_selector.name
+                and printer_name.lower() != printer_selector.name.lower()
             ):
                 continue
 
+            if printer_selector.model_prefix and not printer[
+                "printer-make-and-model"
+            ].lower().startswith(printer_selector.model_prefix.lower()):
+                continue
+
+            if printer_selector.name_prefix and not printer[
+                "printer-info"
+            ].lower().startswith(printer_selector.name_prefix.lower()):
+                continue
```

Comment:
Comment: These are the same checks as previously, just broken into two cases to make it a bit easier to read (since we added a new case)